File: RB120_Object_Oriented_Programming/OOP_Problems/Easy1/Run1/banner_class.rb
class Banner
  def initialize(message)
    @message = message
  end

  def to_s
    [horizontal_rule, empty_line, message_line, empty_line, horizontal_rule].join("\n")
  end

  private

  def horizontal_rule
    get_line('+', '-')
  end

  def empty_line
    get_line('|', ' ')
  end

  def get_line(symbol, padding)
    symbol + padding * (@message.length + 2) + symbol
  end

  def message_line
    "| #{@message} |"
  end
end

banner = Banner.new('To boldly go where no one has gone before.')
puts banner

banner = Banner.new('')
puts banner

File: RB120_Object_Oriented_Programming/OOP_Problems/Easy1/Python/banner_class.py
class Banner:
    def __init__(self, message):
        self._message = message


    def __str__(self):
        print('\n'.join([self._horizontal_rule(), self._empty_line(), self._message_line(), self._empty_line(), self._horizontal_rule()]))


    def _horizontal_rule(self):
        return '+' + '-' * (len(self.message) + 2) + '+'

    def _empty_line(self):
        return '|' + ' ' * (len(self.message) + 2) + '|'

    def _message_line(self):
        return '| ' + self.message + ' |'

    @property
    def message(self):
        return self._message


banner = Banner('To boldly go where no one has gone before.')
print(banner)



